src/EVM/SymExec.hs

```diff
@@ -704,24 +735,87 @@ verify solvers opts preState maybepost = do
           pure (res, leaf)
         let cexs = filter (\(res, _) -> not . isQed $ res) results
         when conf.debug $ putStrLn $ "   Found " <> show (length cexs) <> " potential counterexample(s) in call " <> call
-        pure $ if Prelude.null cexs then (expr, [Qed]) else (expr, fmap toVRes cexs)
+        pure $ (expr, cexs) --]]+ preState expr cexs
   where
-    getCallPrefix :: Expr Buf -> String
-    getCallPrefix (WriteByte (Lit 0) (LitByte a) (WriteByte (Lit 1) (LitByte b) (WriteByte (Lit 2) (LitByte c) (WriteByte (Lit 3) (LitByte d) _)))) = mconcat $ map (printf "%02x") [a,b,c,d]
-    getCallPrefix _ = "unknown"
-    toProps leaf constr post = PNeg (post preState leaf) : constr <> extractProps leaf
-    toPropsFinal leaf constr post = if opts.simp then Expr.simplifyProps $ toProps leaf constr post
-                                                 else toProps leaf constr post
     canBeSat (a, _) = case a of
         [PBool False] -> False
         _ -> True
+
+verifyResults :: VM Symbolic RealWorld -> Expr End -> [(SMTResult, Expr End)] -> (Expr End, [VerifyResult])
+verifyResults preState expr cexs = if Prelude.null cexs then (expr, [Qed]) else (expr, fmap toVRes cexs)
+    where
     toVRes :: (SMTResult, Expr End) -> VerifyResult
     toVRes (res, leaf) = case res of
       Cex model -> Cex (leaf, expandCex preState model)
       Unknown reason -> Unknown (reason, leaf)
       Error e -> Error e
       Qed -> Qed
 
+-- | Symbolically execute the VM and find possible inputs given the
+-- postcondition, if available.
+findInputs
```

Comment:
I don't understand this function... it's the same as `verifyInputs`, except it allows for a fetcher to be passed in, but the only place it's used, the `fetcher` is passed in as `undefined`. We could simply write `verifyInputs` as `findInputs` with the fetcher properly set, right? There seems to be a lot of code duplication here? Or something I am not getting...

Reply:
This was removed as it was not necessary, the verify function was refactored to pass the fetcher.